jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/model/instance/CaseFileInstanceImpl.java

```diff
@@ -113,7 +113,7 @@ public Date getCaseEndDate() {
 
     @Override
     public Map<String, Object> getData() {
-        return this.data;
+        return Collections.unmodifiableMap(this.data);
```

Comment:
unneeded

Reply:
this is to prevent a backdoor update of the intenal map, it should stay to make the solution more robust

Reply:
Change rolled back